feat: doc site (#320)
bentoml/Yatai: docs/source/installation/migration.rst

=========
Migration
=========

Since v1.0.0 has many breaking changes, we need to do some preparation to migrate your existing v0.4.x version of yatai.

.. note:: If your yatai is below v0.4.6, you must upgrade it to v0.4.6 first.

Breaking changes
----------------

* Split into two modules

  * yatai

    Dashboard and bento registry

  * yatai-deployment

    Deploy bento to Kuberenetes

* Remove all yatai component operators

  In v0.4.x, yatai component operators will automatically install dependencies of the component. But in v1.0.0, we removed this feature.

  Some things will changes:

  * No more integration of logging and monitoring

    Because we removed all yatai component operators, yatai now does not automatically integrate logging and monitoring. See the `Observability </observability/index.html>`_ documentation for observability configuration.

> Migration steps

Reviewer: Please clarify if migration requires Yatai system down time. @yetone

---------------

1. Backup PostgreSQL data
"""""""""""""""""""""""""

> .. note:: If you use the external PostgreSQL, you need to skip this step.

Reviewer: ```suggestion
.. note:: If you use an external PostgreSQL, you can to skip this step.
```

Author: @ssheng If using an external database, this must be skipped

.. code:: bash

  pkill kubectl
  kubectl port-forward --namespace yatai-system svc/yatai-postgresql 5433:5432 &
  sleep 6
  PGPASSWORD=$(kubectl -n yatai-system get secret yatai-postgresql -o jsonpath='{.data.postgresql-password}' | base64 -d) pg_dump -h localhost -p 5433 -U postgres -F t yatai > /tmp/yatai.tar

> 2. Get MinIO environment variables

Reviewer: ```suggestion
2. Get object store environment variables
```

""""""""""""""""""""""""""""""""""

> .. note:: If you use the external S3, you need to skip this step.

Reviewer: ```suggestion
.. note:: If you use an external storage such as AWS S3, you can to skip this step.
```

Author: @ssheng  If using an external s3, this must be skipped

.. code:: bash

  export S3_ENDPOINT=minio.yatai-components.svc.cluster.local
  export S3_ACCESS_KEY=$(kubectl -n yatai-components get secret yatai-minio-secret -o jsonpath='{.data.accesskey}' | base64 -d)
  export S3_SECRET_KEY=$(kubectl -n yatai-components get secret yatai-minio-secret -o jsonpath='{.data.secretkey}' | base64 -d)
  export S3_SECURE=false
  export S3_BUCKET_NAME=yatai
  export S3_REGION=i-dont-known

> 3. Test MinIO connection

Reviewer: ```suggestion
3. Test object store connection
```
